### ISSUE TITLE
Issue with wildcard_bin_array

### ISSUE THREAD

**Issue by aneels3** [fvutils/pyvsc#105]
Hi @mballance 
I was trying the wildcard example which is given in the pyvsc documentation.
I have modified it as per my use case.
```python3
import vsc

@vsc.covergroup
class cg(object):

    def __init__(self):
        self.with_sample(
            dict(a=vsc.bit_t(8)))

        self.cp_a = vsc.coverpoint(lambda: self.a[7:0], bins=dict(
            a=vsc.wildcard_bin_array([], "0b1011011x", "0b0x101010")
            ))

obj1 = cg()
```

I am getting the below error
```bash
Traceback (most recent call last):
  File "wildcard.py", line 14, in <module>
    obj1 = cg()
  File "/home/anil/.local/lib/python3.6/site-packages/vsc/coverage.py", line 308, in __init__
    super().__init__(*args, **kwargs)
  File "wildcard.py", line 11, in __init__
    a=vsc.wildcard_bin_array([], "0b1011011x", "0b0x101010")
  File "/home/anil/.local/lib/python3.6/site-packages/vsc/coverage.py", line 521, in __init__
    self.range_l[i][1] = self.range_l[i+1][1]
TypeError: 'tuple' object does not support item assignment

```

How can I specify the different scenarios in a single wildcard_bin_array()?

Thanks and Regards,
Anil

**Comment by aneels3** [comment]
Hi @mballance 
Any inputs on this to make it work?

**Comment by mballance** [comment]
Hi @aneels3,
    This is actually a bug in the wildcard-bin implementation. I've released a fix in 0.5.3 and your testcase runs with that release.

**Comment by aneels3** [comment]
Hi @mballance 
Thanks for the fix.
I am not able to create multiple bins for a single coverpoint.

```python3
import vsc

@vsc.covergroup
class cg(object):

    def __init__(self):
        self.with_sample(
            dict(a=vsc.bit_t(8)))

        self.cp_a = vsc.coverpoint(lambda: self.a[7:0],
            bins=dict(a=vsc.wildcard_bin_array([], "0b1011011x", "0b0x101010")),
            bins=dict(b=vsc.wildcard_bin_array([], "0b1011011x", "0b0x10x01x")))

obj1 = cg()
```

```bash
bins=dict(a=vsc.wildcard_bin_array([], "0b1011011x", "0b0x10x01x")))
    ^
SyntaxError: keyword argument repeated
```

After Changing the bins to bins1 and bins2.
```bash
 bins2=dict(a=vsc.wildcard_bin_array([], "0b1011011x", "0b0x10x01x")))
TypeError: __init__() got an unexpected keyword argument 'bins1'
```

Let me know how can I make this work?

Best,
Anil

**Comment by mballance** [comment]
Hi @aneels3,
    There is a syntax issue with your example above. Only a single 'bins' parameter can be specified per coverpoint. Multiple bins can be defined within that single dict, though. The following works:

```
import vsc

@vsc.covergroup
class cg(object):

    def __init__(self):
        self.with_sample(
            dict(a=vsc.bit_t(8)))

        self.cp_a = vsc.coverpoint(lambda: self.a[7:0],
            bins=dict(
                a=vsc.wildcard_bin_array([], "0b1011011x", "0b0x101010"),
                b=vsc.wildcard_bin_array([], "0b1011011x", "0b0x10x01x"))
        )

obj1 = cg()
```

In looking at this testcase, I uncovered a bin-expansion issue that resulted in too many bins being created. I've corrected that issue in the 0.5.4 release.
